### PR TITLE
FhirClient to common

### PR DESCRIPTION
**Pull request by marcovisserFurore**
Moved the following resources from big common to common:
- `Binary`
- `Bundle`
- `Signature`

Move `FhirClient` from the satellites to common.

Make use of Shared projects

Todo:
- Change codegen to reflect these changes.

See also PR: https://github.com/FirelyTeam/firely-net-common/pull/249

### CODE REVIEW

File: src/Hl7.Fhir.Shared.Core/Model/ModelInfo.cs [hunk 1]
```diff
@@ -310,7 +310,7 @@ public static ModelInspector ModelInspector
             typeof(Model.Distance),
             typeof(Model.Dosage),
             typeof(Model.Duration),
-            typeof(Model.Expression),
+          //  typeof(Model.Expression),
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Shared.Core/Model/ModelInfo.cs:313]: Why is this removed?

**Review by ewoutkramer — CHANGES_REQUESTED** [review]

File: src/Hl7.Fhir.Shared.Core/Model/ModelInfo.cs (hunk 1 above)
Comment (marcovisserFurore) [reply src/Hl7.Fhir.Shared.Core/Model/ModelInfo.cs:313]: Oops, well spotted.

Comment (marcovisserFurore) [reply src/Hl7.Fhir.Shared.Core/Model/ModelInfo.cs]: Solved: `ModelInfo.OpenTypes` has been moved to the satellites.

**Review by ewoutkramer — APPROVED** [review]
